optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/easy/EasyScoreDirector.java

```diff
@@ -16,25 +16,26 @@
 
 package org.optaplanner.core.impl.score.director.easy;
 
-import java.util.Collection;
-
-import org.optaplanner.core.api.domain.solution.Solution;
 import org.optaplanner.core.api.score.Score;
 import org.optaplanner.core.api.score.constraint.ConstraintMatchTotal;
 import org.optaplanner.core.impl.score.director.AbstractScoreDirector;
 import org.optaplanner.core.impl.score.director.ScoreDirector;
 
+import java.util.Collection;
+
 /**
  * Easy java implementation of {@link ScoreDirector}, which recalculates the {@link Score}
- * of the {@link Solution} workingSolution every time. This is non-incremental calculation, which is slow.
+ * of the {@link Solution_} workingSolution every time. This is non-incremental calculation, which is slow.
  * @see ScoreDirector
  */
-public class EasyScoreDirector extends AbstractScoreDirector<EasyScoreDirectorFactory> {
+public class EasyScoreDirector<Solution_>
+        extends AbstractScoreDirector<Solution_, EasyScoreDirectorFactory<Solution_>> {
 
-    private final EasyScoreCalculator easyScoreCalculator;
+    private final EasyScoreCalculator<Solution_> easyScoreCalculator;
 
-    public EasyScoreDirector(EasyScoreDirectorFactory scoreDirectorFactory, boolean constraintMatchEnabledPreference,
-            EasyScoreCalculator easyScoreCalculator) {
+    public EasyScoreDirector(EasyScoreDirectorFactory<Solution_> scoreDirectorFactory,
+                             boolean constraintMatchEnabledPreference,
```

Comment:
Code style: Our code style uses 2 idents for the next line, instead of aligning parameter on the `(` because otherwise long method names create off-page content.